Modules/Scripted/Home/HomeLib/segmentation_model.py

```diff
@@ -60,20 +86,54 @@ def run_inference(self, img):
     if len(img.shape) != 2:
       raise ValueError("img must be a 2D array")
 
-    self.seg_net.eval()
-    img_input = self.transform(img)
-    seg_net_output = self.seg_net(img_input.unsqueeze(0))[0]
-
-    # assumption at the moment is that we have 2-channel image out (i.e. purely binary segmentation was done)
-    assert(seg_net_output.shape[0]==2)
-
-    _, max_indices = seg_net_output.max(dim=0)
-    seg_mask = (max_indices==1).type(torch.uint8)
-
-    model_to_img_matrix = np.diag(np.array(img.shape)/self.image_size)
-
-    return seg_mask, model_to_img_matrix # TODO returning early because post processing causes crash due to ITK python issues
-
-    seg_processed = self.seg_post_process(seg_mask)
+    if self.model_source == self.ModelSource.USE_PTH:
+      self.seg_net.eval()
+      img_input = self.transform(img)
+      seg_net_output = self.seg_net(img_input.unsqueeze(0))[0]
+
+      # assumption at the moment is that we have 2-channel image out (i.e. purely binary segmentation was done)
+      assert(seg_net_output.shape[0]==2)
+
+      _, max_indices = seg_net_output.max(dim=0)
+      seg_mask = (max_indices==1).type(torch.uint8)
+
+      model_to_img_matrix = np.diag(np.array(img.shape)/self.image_size)
+
+      # TODO returning early because post processing causes crash due to ITK python issues
+      # seg_processed = self.seg_post_process(seg_mask)
+      seg_processed = seg_mask
+
+    if self.model_source in (self.ModelSource.USE_LOCAL_ZIP, self.ModelSource.USE_DOCKER_ZIP):
+      # Communicate with monai deploy via files.  Locations are:
+      input_dir_path = tempfile.mkdtemp()
+      output_dir_path = tempfile.mkdtemp()
+      input_file_path = os.path.join(input_dir_path, "input.png")
+      output_mask_path = os.path.join(output_dir_path, "mask.png")
+      output_model_to_img_matrix_path = os.path.join(output_dir_path, "model_to_img_matrix.npy")
+
+      # Write input file
+      img_pil = PIL.Image.fromarray(img)
+      img_pil = img_pil.convert("L")
+      img_pil.save(input_file_path)
+
+      # Run monai-deploy
+      if self.model_source == self.ModelSource.USE_LOCAL_ZIP:
+        deploy_app_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "deploy_app.py")
+        process_image_command = f"monai-deploy exec {deploy_app_path} -m {self.save_zip_path} -i {input_dir_path} -o {output_dir_path}"
+      if self.model_source == self.ModelSource.USE_DOCKER_ZIP:
+        # TODO: Unfortunately, this docker image appears to require CUDA 11.3!!!
+        docker_image = "lung_air_app:latest"
+        # Note: to create the docker image, use
+        # f"monai-deploy package {deploy_app_path} --tag {docker_image} --model {self.save_zip_path}"
+        process_image_command = f"monai-deploy run {docker_image} {input_dir_path} {output_dir_path}"
+      os.system(process_image_command)
```

Comment:
`monai-deploy` is on the PATH with `os.system` but not with `slicer.util.launchConsoleProcess`.  Not even `python` is on the path with the latter.  Rather than tracking down what is wrong there, I am thinking to stick with `os.system`.

Reply:
Sounds good. I see [here](https://github.com/Slicer/Slicer/blob/38e026f75f78320a2cb579b5fa96f0917ab4f02c/Base/Python/slicer/util.py#L3470-L3498) now that when `launchConsoleProcess` is used to execute a python process, one first needs to find `PythonSlicer`... too much trouble.